### DISCUSSION TITLE
how to access vscode webview extension?

### DISCUSSION THREAD

**Discussion by sunchuanstc** [eclipse-theia/theia#14660] [Q&A]
Hi, in local network, in 192.168.1.10 I install theia and export  THEIA_WEBVIEW_EXTERNAL_ENDPOINT="{{hostname}}", and then start theia with self-signed ssl, and in 192.168.1.50  I start chrome with "--ignore-certificate-errors" param  visit the theia,the vscode-webview can normal display.  but if I start chrome without "--ignore-certificate-errors", the webview cannot display!!! I mean, people don't like start chrome with any param, is there another solution to do this?

**Comment by msujew** [comment]
Hey @sunchuanstc,

What kind of error are you getting when starting chrome normally? I'm not sure there's much we can do, since this is a deployment issue.

I would recommend taking a look at tools like [theia-cloud](https://theia-cloud.io/) for deploying a web based Theia app.

Note that changing the webview endpoint to remove the subdomain is a potential security problem, as it allows vscode extensions to extract cookies and other webview data from other extensions.

**Reply by sunchuanstc** [reply]
Hey @msujew , I find the first solution in this issues[   [[]](https://github.com/eclipse-theia/theia/issues/7190)(](https://github.com/eclipse-theia/theia/issues/7190).) I mean,I want start chrome without any param,is there a best solution?

**Reply by msujew** [reply]
> is there a best solution?

Yes, don't use a self-signed certificate. See the [linked stackoverflow answer in the thread you linked](https://stackoverflow.com/a/43484456). Using a self-signed certificate prevents chrome from starting the service worker unless these parameters are specified. In the end, it's a browser/deployment limitation that we cannot work around.
